index.js

```diff
@@ -926,6 +926,82 @@ PersonaClient.prototype.getProfileByGuid = function(opts, callback){
     });
     req.end();
 };
+
+/**
+ * Get all profiles for an array of GUIDs.
+ *
+ * TODO Neither the client lib nor Persona impose restriction on amount of GUIDs requested. Limit in calling app for now.
+ *
+ * @param  {object}   opts     [description]
+ * @param  {array}    opts.guids  Array of GUIDs to fetch profiles for
+ * @param  {string}   opts.token  Auth token
+ * @param  {string}   opts.xRequestId Optional request ID to pass through in logging.
+ * @param  {Function} callback
+ */
+PersonaClient.prototype.getProfilesForGuids = function getProfilesForGuids(opts, callback) {
+    validateOpts(opts,{ guids: _.isArray, token: _.isString });
+    var guids = opts.guids;
+    var token = opts.token;
+    var xRequestId = opts.xRequestId || uuid.v4();
+    var _this = this;
+
+    var ids = '';
+
+    // generate a comma-sep list of user's from the list of users we need to process
+    guids.forEach(function (guid) {
+        ids += (ids !== '') ? ',' + guid : guid;
+    });
```

Comment:
I did wonder that but thought the scenario it was trying to address, rightly or wrongly, was dubious data being passed in like empty values.  TN uses to do this and unless whoever did it didn't realise about join at the time I guess they did it on purpose